server/lib/renderers/doc.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * @description
+ * This library is just a shim to ensure API uniformity with other renderers.
+ * it renders an Word report from html.
+ * @module lib/renderers/xls
+ * @requires juice
+ * @requires q
+ * @requires lodash
+ */
+const juice = require('juice');
+const q = require('q');
+const html = require('./html');
+
+const headers = {
+  'Content-Type' : 'application/vnd.ms-word;charset=utf-8',
+};
+
+exports.render = render;
+exports.extension = '.doc';
+exports.headers = headers;
+
+
+async function render(data, template, options) {
+  const htmlStream = await html.render(data, template, options);
+  const htmlString = juice(htmlStream);
+  const deferred = q.defer();
+  const buffer = Buffer.from(htmlString, 'utf8');
+  deferred.resolve(buffer);
+  return deferred.promise;
```

Comment:
There shouldn't be any need to create a promise here.  Since you are using `async`/`await`, it will automatically return a promise.  I think you can probably just do:
```js
return Buffer.from(htmlString, 'utf8');
```

Reply:
the ReportManager is waiting for a promise

Reply:
You are right.  According to [the MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#Return_value), an asynchronous function returns:

>A Promise which will be resolved with the value returned by the async function, or rejected with an uncaught exception thrown from within the async function.